MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -888,6 +1040,50 @@ void EnggDiffractionViewQtGUI::browseTextureDetGroupingFile() {
   m_uiTabFocus.lineEdit_texture_grouping_file->setText(path);
 }
 
+void EnggDiffractionViewQtGUI::browseFitFocusedRun() {
+  QString prevPath = QString::fromStdString(m_focusDir);
+  if (prevPath.isEmpty()) {
+    prevPath =
+        MantidQt::API::AlgorithmInputHistory::Instance().getPreviousDirectory();
+  }
+  std::string nexusFormat = "Nexus file with calibration table: NXS, NEXUS"
+                            "(*.nxs *.nexus);;";
+
+  QString path(
+      QFileDialog::getOpenFileName(this, tr("Open Focused File "), prevPath,
+                                   QString::fromStdString(nexusFormat)));
+
+  if (path.isEmpty()) {
+    return;
+  }
+
+  MantidQt::API::AlgorithmInputHistory::Instance().setPreviousDirectory(path);
+  m_uiTabFitting.lineEdit_pushButton_run_num->setText(path);
+}
+
+void EnggDiffractionViewQtGUI::browsePeaksToFit() {
+  QString prevPath = QString::fromStdString(m_calibSettings.m_inputDirRaw);
+  if (prevPath.isEmpty()) {
+    prevPath =
+        MantidQt::API::AlgorithmInputHistory::Instance().getPreviousDirectory();
+  }
+
+  QString path(
+      QFileDialog::getOpenFileName(this, tr("Open Peaks To Fit"), prevPath,
+                                   QString::fromStdString(g_DetGrpExtStr)));
+
+  if (path.isEmpty()) {
+    return;
+  }
+
+  MantidQt::API::AlgorithmInputHistory::Instance().setPreviousDirectory(path);
+
+  std::string peaksData = readPeaksFile(path.toStdString());
```

Comment:
This function `readPeaksFile` could throw if there are issues when reading the file (for example you try to read a file that is not readable, etc.). This is called from a SLOT which implies danger of bad crash. I'd put it inside a `try/catch`.